Content.Shared/_NC/Trade/Components/NcStoreComponent.cs

```diff
@@ -0,0 +1,62 @@
+using Robust.Shared.GameStates;
+
+
+namespace Content.Shared._NC.Trade;
+
+
+public readonly record struct StoreListingKey(StoreMode Mode, string ListingId);
+
+[RegisterComponent, NetworkedComponent]
+public sealed partial class NcStoreComponent : Component
+{
+    public int CatalogRevision;
+    public EntityUid? CurrentUser;
+    public int UiRevision;
+
+    [ViewVariables]
+    public HashSet<string> CompletedOneTimeContracts { get; } = new();
+
+    [DataField("categories")]
+    public List<string> Categories { get; set; } = new();
+
+    [DataField("currencyWhitelist")]
+    public List<string> CurrencyWhitelist { get; set; } = new();
+
+    public List<NcStoreListingDef> Listings { get; set; } = new();
+
+    [ViewVariables]
+    public Dictionary<StoreListingKey, NcStoreListingDef> ListingIndex { get; } = new();
+
+    [DataField("buyPresets")]
+    public List<string> BuyPresets { get; set; } = new();
+
+    [DataField("sellPresets")]
+    public List<string> SellPresets { get; set; } = new();
+
+    [DataField("contractPresets")]
```

Comment:
```suggestion
    [DataField]
```
Можно просто датафилд указать, так как оно совпадает с названием переменной